Controller.php

```diff
@@ -114,6 +115,35 @@ public function deleteClientCredentials()
         return $this->index();
     }
 
+	private function setFixedEndDateIfWordPress($endDate)
+	{
+		// if Matomo for WordPress is used, then the data will be imported into the same site as the data is also being
+		// tracked into by the sounds. So we need to make sure the import ends before Matomo for WordPress was installed
+		// otherwise it would potentially always overwrite already aggregated report data
+		if (method_exists(SettingsServer::class, 'isMatomoForWordPress')
+		    && SettingsServer::isMatomoForWordPress()
+		    && defined('\WpMatomo\Installer::OPTION_NAME_INSTALL_DATE')) {
+
+			$installDate = get_option(\WpMatomo\Installer::OPTION_NAME_INSTALL_DATE);
+
+			if (empty($installDate)) {
+				// matomo for WordPress was installed before this option was set
+				// we have to make sure there will be an end date otherwise it will always overwrite data
+				// we assume Matomo for WordPress was installed two days ago
+				$installDate = Date::today()->subDay(2);
```

Comment:
this should probably set the option too right? otherwise if a user imports at a later date (to re-import data for some reason), it will overwrite tracked data

Reply:
@diosmosis very good point. I won't be changing this for now as I don't want to store a wrong value for `install_date`. I suppose it's bit of an edge case and unlikely anyone would run into this (or too many people would run into this) since the install date was only missing in some older versions of Matomo for WordPress and not too many use the importer.

Reply:
actually I will then use the site creation date

Reply:
Wouldn't that be before the install date though? Assuming there's data to import from before the site was created in matomo.

Reply:
Actually, the importer might need to be able to move the site creation time back when importing before that date...

Reply:
Good point. Changed it back to original logic